x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStopTrainedModelDeploymentAction.java

```diff
@@ -114,79 +115,64 @@ protected void doExecute(
-                    listener.onFailure(
-                        new ElasticsearchStatusException(
-                            "Cannot stop deployment for model [{}] as it has a model_alias [{}] that is still referenced"
-                                + " by ingest processors; use force to stop the deployment",
-                            RestStatus.CONFLICT,
-                            modelId,
-                            referencedModelAlias.get()
-                        )
-                    );
-                    return;
-                }
+            List<String> modelAliases = getModelAliases(state, request.getId());
+            Optional<String> referencedModelAlias = modelAliases.stream().filter(referencedModels::contains).findFirst();
+            if (referencedModelAlias.isPresent()) {
+                listener.onFailure(
+                    new ElasticsearchStatusException(
+                        "Cannot stop deployment [{}] as it has a model_alias [{}] that is still referenced"
+                            + " by ingest processors; use force to stop the deployment",
+                        RestStatus.CONFLICT,
+                        request.getId(),
+                        referencedModelAlias.get()
+                    )
+                );
+                return;
             }
+        }
 
-            // NOTE, should only run on Master node
-            assert clusterService.localNode().isMasterNode();
-            trainedModelAssignmentClusterService.setModelAssignmentToStopping(
-                modelId,
-                ActionListener.wrap(
-                    setToStopping -> normalUndeploy(task, models.get(0).getModelId(), maybeAssignment.get(), request, listener),
-                    failure -> {
-                        if (ExceptionsHelper.unwrapCause(failure) instanceof ResourceNotFoundException) {
-                            listener.onResponse(new StopTrainedModelDeploymentAction.Response(true));
-                            return;
-                        }
-                        listener.onFailure(failure);
+        // NOTE, should only run on Master node
+        assert clusterService.localNode().isMasterNode();
+        trainedModelAssignmentClusterService.setModelAssignmentToStopping(
+            request.getId(),
+            ActionListener.wrap(
+                setToStopping -> normalUndeploy(task, request.getId(), maybeAssignment.get(), request, listener),
+                failure -> {
+                    if (ExceptionsHelper.unwrapCause(failure) instanceof ResourceNotFoundException) {
+                        listener.onResponse(new StopTrainedModelDeploymentAction.Response(true));
+                        return;
                     }
-                )
-            );
-        }, listener::onFailure);
-
-        GetTrainedModelsAction.Request getModelRequest = new GetTrainedModelsAction.Request(request.getId(), null, Collections.emptySet());
```

Comment:
Previously there was a check that the model config exists before trying to stop a deployment. Getting the model config is not necessary to stop the deployment the fact that the deployment exists is enough